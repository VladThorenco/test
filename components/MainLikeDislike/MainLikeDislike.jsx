import './MainLikeDislike.scss';
import Like from '__test__/MainPage/Like';
import PropTypes from 'prop-types';
import React from 'react';

const propTypes = {
    /* passed props */
    likedId: PropTypes.string.isRequired,
    likedType: PropTypes.string.isRequired,
};

const MainLikeDislike = (props) => {
    const {likedId, likedType } = props;
    return (
        <div className="container_likes">
            <Like
                likedId={likedId}
                likedType={likedType}
                isDislike={false}
                typeLike='liked'
                typeCount='likes_count'
                actionName='like'
            />
            <Like
                likedId={likedId}
                likedType={likedType}
                isDislike={true}
                typeLike='disliked'
                typeCount='dislikes_count'
                actionName='dislike'
            />
        </div>
    );
};

MainLikeDislike.propTypes = propTypes;

export default MainLikeDislike;
