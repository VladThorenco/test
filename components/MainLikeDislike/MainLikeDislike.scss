.container_likes {
  display: flex;
}
