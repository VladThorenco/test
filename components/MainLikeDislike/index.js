import MainLikeDislike from './MainLikeDislike';

export default MainLikeDislike;
