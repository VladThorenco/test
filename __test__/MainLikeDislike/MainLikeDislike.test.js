import {shallowWithoutStore, mountWithoutStore} from '../../../shallowWrapper';
import MainLikeDislike from '__test__/MainPage/MainLikeDislike';
import React from 'react';
import {threePosts} from 'data/postData';

jest.mock('__test__/MainPage/Like', () => () => <div id="Like"/>);

describe('MainLikeDislike', () => {
    const props = {
        likedId: 'abcdf-123-fgt-2',
        likedType: 'Post',
    };

    it('renders without crashing', () => {
        const component = shallowWithoutStore(<MainLikeDislike {...props} likedId={threePosts[0].id} likedType='Post'/>);
        expect(component).toMatchSnapshot();
    });

    it('renders component Like', () => {
        const component = mountWithoutStore(<MainLikeDislike {...props} likedId={threePosts[0].id} likedType='Post'/>);
        const thisComponent = component.find('MainLikeDislike').find('#Like');
        expect(thisComponent).toHaveLength(2);
    });
});
