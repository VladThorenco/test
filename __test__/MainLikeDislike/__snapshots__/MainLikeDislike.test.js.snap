// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLikeDislike renders without crashing 1`] = `ShallowWrapper {}`;
